runtime/near-vm-runner/src/runner.rs

```diff
@@ -30,6 +31,7 @@ pub fn run<'a>(
     promise_results: &'a [PromiseResult],
     current_protocol_version: ProtocolVersion,
     cache: Option<&'a dyn CompiledContractCache>,
+    in_mem_contract: InMemoryContracts,
```

Comment:
General observation: function with many arguments are also functions with frequently changing set of arguments. We should look into packing some of the args here into a struct, to reduce code churn:

https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#functions-with-many-parameters

but this is unrelated to the PR

Reply:
Will do in later refactor PRs, you're also welcome to do, your refactor PRs look very good